### ISSUE TITLE
add wireframe to Surface layer

### ISSUE THREAD

**Issue by kevinyamauchi** [napari/napari#2970]
## 🚀 Feature
It looks like `vispy` 0.7 added wireframe support to the `MeshVisual`.

See the tweet [here](https://twitter.com/vispyproject/status/1410586730978320386) and an example [here](https://github.com/vispy/vispy/blob/main/examples/basics/scene/mesh_shading.py).

## Motivation

Viewing the wireframe can make it easier to inspect the mesh.

## Pitch

We attach a `WireFrameFilter` to the `MeshVisual` node and then expose the `WireFrameFilter.enabled` as a `bool` property on the `Surface` model.

## Alternatives
We could vendor `WireFrameFilter` for now and then switch to `vispy` until we want to require  `vispy >= 0.7.0`

## Additional context
Requires `vispy >= 0.7.0`

**Comment by kevinyamauchi** [comment]
This would be super useful to my work - I am happy to implement it if there is interest.

**Comment by alisterburt** [comment]
I'm definitely interested and think this would be super useful 🙂 

I would like to easily access the following states for the `WireFrameFilter`

```python
dict(wireframe_only=False, faces_only=False,),
dict(wireframe_only=True, faces_only=False,)
```

To be able to both overlay the wireframe on the existing mesh and also show only the wireframe. 

In terms of API, what about having a `layer.wireframe` attribute on the surface layer which could be
- `None`
- `overlay`
- `wireframe_only`

Having those options in a `WireFrame` enum and available via a dropdown on the surface layer?

In a followup PR we could also consider exposing the face normals with the new [`MeshNormalsVisual`](https://github.com/vispy/vispy/blob/22b9355f580f213856a3b335e8fd56650457fed6/vispy/visuals/mesh_normals.py)
![image](https://user-images.githubusercontent.com/7307488/124301616-fbea3500-db57-11eb-8222-2f06a14b0e38.png)

**Comment by tlambert03** [comment]
it's a hairy triceratops!

**Comment by alisterburt** [comment]
[not even fake news!](https://www.news.com.au/technology/science/new-fossil-suggests-dinosaurs-may-have-all-been-hairy/news-story/4a51c610d0daced90beb7e2719dac252?sv=b57b7699d6702965c935437dc9f10170)

**Comment by kevinyamauchi** [comment]
> In terms of API, what about having a layer.wireframe attribute on the surface layer which could be
>
>    - None
>  -  overlay
>   - wireframe_only

This makes sense to me!

**Comment by sofroniewn** [comment]
I believe this can be closed by #3689 right @kevinyamauchi @alisterburt @brisvag?

**Comment by alisterburt** [comment]
Yep! Closing
